src/org/zaproxy/zap/extension/pscanrulesAlpha/FeaturePolicyScanner.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ * 
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ * 
+ * Copyright 2019 The ZAP Development Team
+ *  
+ * Licensed under the Apache License, Version 2.0 (the "License"); 
+ * you may not use this file except in compliance with the License. 
+ * You may obtain a copy of the License at 
+ * 
+ *   http://www.apache.org/licenses/LICENSE-2.0 
+ *   
+ * Unless required by applicable law or agreed to in writing, software 
+ * distributed under the License is distributed on an "AS IS" BASIS, 
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. 
+ * See the License for the specific language governing permissions and 
+ * limitations under the License. 
+ */
+
+package org.zaproxy.zap.extension.pscanrulesAlpha;
+
+import java.util.Vector;
+
+import org.apache.log4j.Logger;
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.core.scanner.Alert;
+import org.parosproxy.paros.network.HttpMessage;
+import org.zaproxy.zap.extension.pscan.PassiveScanThread;
+import org.zaproxy.zap.extension.pscan.PluginPassiveScanner;
+
+import net.htmlparser.jericho.Source;
+
+/**
+ * Feature Policy Header Missing passive scan rule 
+ * https://github.com/zaproxy/zaproxy/issues/4885
+ */
+
+public class FeaturePolicyScanner extends PluginPassiveScanner {
+
+	private static final String MESSAGE_PREFIX = "pscanalpha.featurepolicymissing.";
+	private static final int PLUGIN_ID = 10063;
+	
+	private PassiveScanThread parent = null;
+	private static final Logger logger = Logger.getLogger(FeaturePolicyScanner.class);
+	
+
+	@Override
+	public void scanHttpRequestSend(HttpMessage httpMessage, int id) {
+		// Only checking the response for this plugin
+	}
+
+	@Override
+	public void scanHttpResponseReceive(HttpMessage httpMessage, int id, Source source) {
+		long start = System.currentTimeMillis();
+		
+		if (! httpMessage.getResponseHeader().isHtml() &&
+				! httpMessage.getResponseHeader().isJavaScript()) {
+			// Only really applies to HTML and Javascript responses, but also check on Low threshold
+			return;
+		}
+		
+		//Get the Feature Policy headers
+		boolean featurePolicyHeaderFound = false;
+				
+		//Feature-Policy is supported by Chrome 55+, Firefox 65+, Opera 43+, but not by Internet Exploder
+		Vector<String> featurePolicyOptions = httpMessage.getResponseHeader().getHeaders("Feature-Policy");
+		//If it's not null or empty then we found one
+		if (featurePolicyOptions != null && featurePolicyOptions.isEmpty() == false) { 
+			featurePolicyHeaderFound = true;
```

Comment:
The assignment could be done directly without the if, or even just be skipped in favour of a single if removing this and replacing the next. (The intermediate storage isn’t really necessary).

I know this was likely copied from an example....